concepts/05 UI Components/zz Common/30 Templates/30 3rd-Party Template Engines.md

```diff
@@ -1,34 +1,15 @@
-You can use a 3rd-party template engine, but only with jQuery. DevExtreme supports the following template engines out of the box:
+You can use the <a href="http://mustache.github.io/" target="_blank">Mustache</a> 3rd-party template engine. Call the [DevExpress.setTemplateEngine(options)](/api-reference/50%20Common/utils/setTemplateEngine(options).md '/Documentation/ApiReference/Common/Utils/#setTemplateEngineoptions') method and implement functions that `compile` and `render` templates:
 
-- <a href="http://underscorejs.org/" target="_blank">Underscore</a>
-- <a href="https://github.com/BorisMoore/jsrender" target="_blank">JsRender</a>
-- <a href="http://mustache.github.io/" target="_blank">Mustache</a>
-- <a href="http://twitter.github.io/hogan.js/" target="_blank">Hogan</a>
-- <a href="http://handlebarsjs.com/" target="_blank">Handlebars</a>
-- <a href="http://olado.github.io/doT/index.html" target="_blank">doT</a>
+    <!-- tab: index.js -->
+    DevExpress.setTemplateEngine({
+        compile: (element) => $(element).html(),
+        render: (template, data) => Mustache.render(template, data),
+    });
 
-To use one of them, pass its name to the [DevExpress.setTemplateEngine(name)](/api-reference/50%20Common/utils/setTemplateEngine(name).md '/Documentation/ApiReference/Common/Utils/#setTemplateEnginename') method:
+If you assign a function that returns a string to a template, the `compile` callback receives a text node as an argument. In this case, the `compile` callback needs to contain additional logic that determines if an argument is a template element or a text node:
```

Comment:
[minor] This is minor, but I'd like to be sure we are on the same page here.

The colon and the bullet list formatting here make an impression like the list describes the "additional logic" mentioned in the previous sentence.

However, that's not what we mean.

The users need to:
1. Implement additional logic to determine if an argument is a template element or a text node (we do not give details on it)
2. Based on the result of the previous step, either do `$(element).html()` or `$('<div>').append(element).html()`